File: backend/src/main/java/ch/morgias/cookgenda/models/agenda/PlanedRecipe.java
package ch.morgias.cookgenda.models.agenda;

import ch.morgias.cookgenda.models.food.Recipe;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
public class PlanedRecipe {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    private Recipe recipe;
    private LocalDateTime date;
}

File: android/app/src/main/java/ch/morgias/cookgenda/android/models/PlanedRecipe.kt
package ch.morgias.cookgenda.android.models

import ch.morgias.cookgenda.android.models.utils.LocalDateTimeSerializer
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable
import java.time.LocalDateTime

@Serializable
data class PlanedRecipe(
    val recipeId: Int,
    val planedRecipeId: Int,
    val name: String,
    @Serializable(with = LocalDateTimeSerializer::class) @Contextual val date: LocalDateTime
)